src/Compilers/Core/Portable/Compilation/Expression.cs

```diff
@@ -0,0 +1,365 @@
+        {
+            switch (type.SpecialType)
+            {
+                case SpecialType.System_Boolean:
+                    return Semantics.LiteralKind.Boolean;
+                case SpecialType.System_Byte:
+                case SpecialType.System_Int16:
+                case SpecialType.System_Int32:
+                case SpecialType.System_Int64:
+                case SpecialType.System_IntPtr:
+                case SpecialType.System_SByte:
+                case SpecialType.System_UInt16:
+                case SpecialType.System_UInt32:
+                case SpecialType.System_UInt64:
+                case SpecialType.System_UIntPtr:
+                    return Semantics.LiteralKind.Integer;
+                case SpecialType.System_DateTime:
+                    return Semantics.LiteralKind.DateTime;
+                case SpecialType.System_Decimal:
+                    return Semantics.LiteralKind.Decimal;
+                case SpecialType.System_Double:
+                case SpecialType.System_Single:
+                    return Semantics.LiteralKind.Floating;
+                case SpecialType.System_Char:
+                    return Semantics.LiteralKind.Character;
+                case SpecialType.System_String:
+                    return Semantics.LiteralKind.String;
+            }
+
+            return Semantics.LiteralKind.None;
+        }
+    }
+
+    public class Relational : IRelational
+    {
+        public Relational(RelationalOperationKind relationalKind, IExpression left, IExpression right, ITypeSymbol resultType, IMethodSymbol operatorMethod, SyntaxNode syntax)
+        {
+            this.RelationalKind = relationalKind;
+            this.Left = left;
+            this.Right = right;
+            this.ResultType = resultType;
+            this.Operator = operatorMethod;
+            this.Syntax = syntax;
+        }
+
+        public RelationalOperationKind RelationalKind { get; private set; }
+        
+        public IExpression Left { get; private set; }
+        
+        public IExpression Right { get; private set; }
+        
+        public SyntaxNode Syntax { get; private set; }
+
+        public bool UsesOperatorMethod => this.Operator != null;
+
+        public IMethodSymbol Operator { get; private set; }
+
+        public ITypeSymbol ResultType { get; private set; }
+
+        public OperationKind Kind => OperationKind.RelationalOperator;
+
+        public object ConstantValue => null;
+    }
+
+    public class ConditionalChoice : IConditionalChoice
+    {
+        public ConditionalChoice(IExpression condition, IExpression ifTrue, IExpression ifFalse, ITypeSymbol resultType, SyntaxNode syntax)
+        {
+            this.Condition = condition;
+            this.IfTrue = ifTrue;
+            this.IfFalse = ifFalse;
+            this.ResultType = resultType;
+            this.Syntax = syntax;
+        }
+
+        public IExpression Condition { get; private set; }
```

Comment:
"private set; }"  could be removed.  unless these are intended to be assigned after the construtor.